front/src/applications/stdcmV2/components/StdcmConsist.tsx

```diff
@@ -0,0 +1,154 @@
+import { type LightRollingStockWithLiveries } from 'common/api/osrdEditoastApi';
+import { type SelectOptionObject } from 'common/BootstrapSNCF/SelectSNCF';
+import { useOsrdConfActions, useOsrdConfSelectors } from 'common/osrdContext';
+import SpeedLimitByTagSelector from 'common/SpeedLimitByTagSelector/SpeedLimitByTagSelector';
+import { useStoreDataForSpeedLimitByTagSelector } from 'common/SpeedLimitByTagSelector/useStoreDataForSpeedLimitByTagSelector';
+import RollingStock2Img from 'modules/rollingStock/components/RollingStock2Img';
+import useFilterRollingStock from 'modules/rollingStock/components/RollingStockCard/useFilterRollingStock';
+import type { StdcmConfSliceActions } from 'reducers/osrdconf/stdcmConf';
+import type { StdcmConfSelectors } from 'reducers/osrdconf/stdcmConf/selectors';
+import { useAppDispatch } from 'store';
+
+import StdcmCard from './StdcmCard';
+import StdcmSuggestions from './StdcmSuggestions';
+
+interface StdcmSuggestionsConsistOption
+  extends SelectOptionObject,
+    Omit<LightRollingStockWithLiveries, 'id'> {
+  value: LightRollingStockWithLiveries;
+}
+
+function ConsistCardTitle({
+  rollingStock,
+}: {
+  rollingStock?: LightRollingStockWithLiveries | null;
+}) {
+  if (!rollingStock) return null;
+
+  return (
+    <div className="consist-img w-50 d-flex justify-content-end">
+      <RollingStock2Img rollingStock={rollingStock} />
+    </div>
+  );
+}
+
+export default function StdcmConsist({ isPending = false }: { isPending?: boolean }) {
+  const { speedLimitByTag, speedLimitsByTags, dispatchUpdateSpeedLimitByTag } =
+    useStoreDataForSpeedLimitByTagSelector();
+
+  const { updateRollingStockID } = useOsrdConfActions() as StdcmConfSliceActions;
+  const dispatch = useAppDispatch();
+
+  // TODO: Handle error
+  const { data: { results: rollingStocks } = { results: [] }, isSuccess } =
+    enhancedEditoastApi.endpoints.getLightRollingStock.useQuery({
+      pageSize: 1000,
+    });
+
+  const { getRollingStockID } = useOsrdConfSelectors() as StdcmConfSelectors;
+  const rollingStockID = useSelector(getRollingStockID);
+  const selectedRs = rollingStocks.find((rs) => rs.id === rollingStockID);
+
+  const [filteredRollingStockList, setFilteredRollingStockList] =
+    useState<LightRollingStockWithLiveries[]>(rollingStocks);
+  const [isSelectChanged, setIsSelectChanged] = useState(false);
+
+  const { filters, searchMateriel } = useFilterRollingStock({
+    isSuccess,
+    rollingStocks,
+    filteredRollingStockList,
+    setFilteredRollingStockList,
+  });
+
+  const getLabel = (rs: LightRollingStockWithLiveries) => {
+    let res = '';
+    const { metadata, name } = rs;
+
+    const series = metadata?.series ?? (metadata?.reference || '');
+    const subseries =
+      metadata?.series && metadata.series !== metadata.subseries
+        ? metadata.subseries
+        : metadata?.detail || '';
+
+    if (series) res += series;
+    if (subseries) res += series ? ` (${subseries})` : subseries;
```

Comment:
Add a test 'if (metadata)', also clean the tests for metadata (remove the ? and the || '')
```suggestion
    if (metadata) {
      const series = metadata.series ?? metadata.reference;
      const subseries =
        metadata.series && metadata.series !== metadata.subseries
          ? metadata.subseries
          : metadata.detail;

      if (series) res += series;
      if (subseries) res += series ? ` (${subseries})` : subseries;
    }
```